units/schedule_patch.go

```diff
@@ -67,3 +85,34 @@ func SchedulePatch(ctx context.Context, env evergreen.Environment, patchId strin
 	}
 	return http.StatusOK, nil
 }
+
+// addPreviousSingleHostTaskGroupTasks checks all the tasks in a task group in the patch update
+// and if that task group is a single host task group, it schedules the previous tasks in the
+// task group.
+func addPreviousSingleHostTaskGroupTasks(patchUpdateReq *model.PatchUpdate, project *model.Project) {
+	taskGroupTasksToAddToVariant := map[string]string{}
+	for _, vt := range patchUpdateReq.VariantsTasks {
+		for _, t := range vt.Tasks {
+			tg := project.FindTaskGroupForTask(t)
```

Comment:
Can this pass the bv name as well? Would make traversal more efficient/less ambiguous. Since task groups can be defined inline for a variant and those take precedent, just looking up a task by name is not unique enough to identify which task group the task belongs to.